File: src/types/index-segment.ts
import type { KnowledgeMetadata, KnowledgeObjectType, SourceRef, WikiIndexStatus } from "@/types/wiki"

export type IndexSegment = {
  segmentId: string
  nodeId: string
  nodeTitle: string
  objectType?: KnowledgeObjectType
  subtype?: string
  segmentType:
    | "title"
    | "summary"
    | "body"
    | "section"
    | "table"
    | "qa"
    | "metadata"
    | "condition"
    | "procedure_step"
    | "troubleshooting_step"
  content: string
  title?: string
  contentPreview: string
  tokenCount: number
  enabled: boolean
  indexStatus: WikiIndexStatus
  vectorDocId?: string
  lastIndexedAt?: string
  retrievalHits: number
  avgScore?: number
  sourceRefs: SourceRef[]
  sourceRefIds?: string[]
  processingProfile?: string
  metadataSummary?: Array<{
    label: string
    value: string
  }>
  createdAt?: string
  updatedAt?: string
  metadata: {
    nodeType: string
    status: string
    tags: string[]
  } & KnowledgeMetadata
}

File: src/main/java/com/wikinode/studio/model/IndexSegment.java
package com.wikinode.studio.model;

import java.util.List;
import java.util.Map;

public record IndexSegment(
  String segmentId,
  String nodeId,
  String nodeTitle,
  String objectType,
  String subtype,
  String segmentType,
  String content,
  String title,
  String contentPreview,
  int tokenCount,
  boolean enabled,
  String indexStatus,
  String vectorDocId,
  String lastIndexedAt,
  int retrievalHits,
  Double avgScore,
  List<SourceRef> sourceRefs,
  List<String> sourceRefIds,
  String processingProfile,
  List<IndexSegmentMetadataSummaryItem> metadataSummary,
  String createdAt,
  String updatedAt,
  Map<String, Object> metadata
) {
}
